[Reader] Show "Create Post" FAB in Reader

RenanLukas opened the pull request:
Fixes #20908 

- Implement `reader_floating_button` remote feature flag
- Show the "create" FAB in Reader based on the state of the `reader_floating_button` feature flag
- Add analytics (see #20908 for details)
- Move some FAB/Sheet related code to `MainCreateSheetHelper` for better organization
- Some minor refactor

-----

## To Test:

### Showing FAB on Reader page
1. Open Jetpack
2. Go to `Debug Settings`
3. **Disable** the `reader_floating_buitton` remote flag
4. Go to Reader
5. **Verify** the FAB does not appear on Reader
6. Go to `Debug Settings`
7. **Enable** the `reader_floating_button` remote flag
8. Go to Reader
9. **Verify** the FAB appears on Reader

### Analytics
1. Open Jetpack
2. Go to `Debug Settings`
3. **Enable** the `reader_floating_button` remote flag
4. Go to Reader
5. **Verify** tracked `reader_create_fab_shown`
6. Tap the FAB
7. **Verify** tracked `reader_create_sheet_shown`
8. Tap the "?" button in the Prompts item (if a prompt is available for the tested site)
9. **Verify** tracked `reader_create_sheet_prompt_help_tapped`
10. Go back
11. Tap `Answer prompt` (if a prompt is available for the tested site)
12. **Verify** tracked `reader_create_sheet_answer_prompt_tapped`
13. Go back to Reader
14. Tap the FAB
15. Tap `Blog post`
16. **Verify** tracked `reader_create_sheet_action_tapped, Properties: {"action":"create_new_post"}`

### Sanity
Please also test the FAB works normally on "My Site" page as expected, and the FAB is not shown in other pages, nor on the Reader when the `reader_floating_button` is disabled.

-----

## Regression Notes

1. Potential unintended areas of impact

    - FAB behavior in other pages

18. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual Tests
    - Unit tests in the Activity ViewModel

19. What automated tests I added (or what prevented me from doing so)

    - Updated Activity ViewModel tests
    - Added unit tests to the `MainCreateSheetHelper`

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

Commit:
Add create post FAB to Reader

Commit:
Show correct message based on current page

Commit:
Add showFabForPage helper function

This is where logic should be added if we need to also limit specific scenarios
for showing the button in a specific page (such as a feature flag).

Commit:
Create FeatureConfig for ReaderFloatingButton

Commit:
Move some FAB and content creation sheet logic to Helper

Commit:
Use Helper in ViewModel for some FAB logic

Commit:
Add/update create sheet and button analytics

Commit:
Use SiteUtilsWrapper for better testability

Commit:
Add unit tests for MainCreateSheetHelper

Commit:
Add/Update unit tests for WPMainActivityViewModel

dangermattic commented:
<!--
  0 Errors
  1 Warning: PR is not assigned to a milest...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

RenanLukas commented:
Code already reviewed on https://github.com/wordpress-mobile/WordPress-Android/pull/20910

Commit:
Change coordinator ID in filtered_list_component

This layout is usually inflated in Reader feeds inside the Main Activity of the
app, which already contains a View with the same ID that's used for snackbar
and properly positioning the FAB, which is now visible in Reader.

This was causing a bug where the snackbar was shown behind the bottom bar
instead of pushing the FAB up and being shown between the bottom bar and FAB.

Commit:
Extract selected page logic into method

Commit:
Create tracker for MainCreateSheet

Commit:
Replace helper with tracker for MainCreateSheet analytics

thomashorta reviewed (approved):
Approving it since Renan already reviewed this code (which was originally made by me on a different PR) but since he opened this PR he can't approve it again.
